rfcs/RFC-Uploads.md

```diff
@@ -0,0 +1,356 @@
+# User Uploads
+
+## For the cellxgene Data Portal
+
+**Authors:** [Trent Smith](mailto:trent.smith@chanzuckerberg.com)
+
+**Approvers:** [Friendly Tech Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly PM Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly Tech Person2](mailto:some.nice.person@chanzuckerberg.com), [Girish Patangay](mailto:girish.patangay@chanzuckerberg.com), [Sara Gerber](mailto:sara.gerber@chanzuckerberg.com)
+
+## tl;dr
+
+A user can upload files of any size to Data Portal using either a shared link from their cloud provider, or directly from their browser.
+
+## Glossary
+
+- CSP(Cloud Storage Provider) - Cloud storage services such as Dropbox or Google Drive.
+- Shared Link - A link from a CSP that give the bearer of that link access to private data.
+- DP(Data Portal) - The cellxgene Data Portal.
+- Small File - a file < 5 GB in size.
+- Large File - a file > 5 GB in size.
+
+## Problem Statement | Background
+
+One of the primary goals of the DP is to allow users to upload their own data sets. These data sets can vary in
+size from a few MB to several GB. User may want to upload from their local machine or from their CSP. For
+users that want to upload from their local machine, they must be able to resume an upload in the event their connection
+is interrupted. For users that want to upload data sets from their CSP, they are able to provide a shared link from
+their CSP that is used to upload directly to S3.
+
+## Product Requirements
+
```

Comment:
Should there be a requirement around rejecting CSP links that point to more than 1 file based on some of the details in the implementation section?

Reply:
This is tricky to discern. If a shared link refers to more than one file, both Dropbox and Google Drive, automatically package them up in a .zip. So to us it would appear to be one file, until we inspect it and see that it's a .zip. I'll need to check and see if we can make that interpretation before queuing it up.

Reply:
I added that as a requirement